### PR TITLE
ManageIQ module - provider refresh

### PR DESCRIPTION
**Pull request by abraverm**
##### SUMMARY
ManageIQ provider API allows refershing, which tells ManageIQ to get information from the provider.  This action is sometimes needed in daily operations and management process, for example if ManageIQ will loose connection to the provider over long period and the operator fixed the issue, then refreshing is needed.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
manageiq-proviedr

##### ANSIBLE VERSION
```
ansible 2.6.0 (manageiq_provider_refresh 57f2907ad6) last updated 2018/03/18 14:43:40 (GMT +300)
  config file = /home/abraverm/.ansible.cfg
  configured module search path = [u'/home/abraverm/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/abraverm/contribute/ansible/lib/ansible
  executable location = /home/abraverm/contribute/ansible/bin/ansible
  python version = 2.7.14 (default, Feb 27 2018, 20:43:24) [GCC 7.3.1 20180130 (Red Hat 7.3.1-2)]
```

##### ADDITIONAL INFORMATION
http://manageiq.org/docs/reference/latest/api/reference/providers#refresh-providers

### CODE REVIEW

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py [hunk 1]
```diff
@@ -771,12 +771,28 @@ def create_provider(self, name, provider_type, endpoints, zone_id, provider_regi
             changed=True,
             msg="successfully created the provider %s: %s" % (name, result['results']))
 
+    def refresh(self, provider, name):
+        """ Trigger provider refresh.
+
+        Returns:
```
Comment (yaacov) [inline lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:777]: what does the `result['results']` object contain on successful refresh  ?

**Comment by yaacov** [comment]
shipit

**Comment by cben** [comment]
Nice!
shipit

**Comment by cben** [comment]
If provider doesn't exist, and user specified `state: refresh`, should module try to both create & refresh?
@abraverm @yaacov what do you think ?

**Comment by yaacov** [comment]
@cben my vote is for explicitly do "present" , but it does make seance that "refresh" imply "present" ... , 

p.s.
@abraverm , I will "shipit" if you make a PR that adds a "refresh" flag to create, so we can make "create" with, or without "refresh" (and then we can remove the "refresh" state, because it's not really a state like "absent" and "present", it's only a flag of "present" )

**Comment by cben** [comment]
refreshing is an action, not exactly an idempotent state.  Running the
module 3 times may trigger up to 3 refreshes.

IMHO, hiding an action in `state: refresh` is acceptable misuse and kinda
cute, while hiding an action in `state: present, refresh: true` would be
too far from the truth.
I can't fully explain why, this is just a feeling (and deep bikeshedding),
so I'll go along with whatever.

**Comment by abraverm** [comment]
I agree with @yaacov on explicit "present", and I will add that "present" requires many arguments while "refresh" is just a trigger.

Regarding state, maybe the right wording should be "refreshed" and that I think is acceptable state of a provider. The problem is that refreshing has to end for calling it  "refreshed" but that can take a very long time, so trigerring is enough IMHO and  then "refresh" is more correct in this case.

**Comment by yaacov** [comment]
If you do something like that I will "shipit" :-) but what we have now is ok by me too :+1:

**Pull request merged**